demos/human_pose_estimation_demo/main.cpp

```diff
@@ -74,50 +79,138 @@ int main(int argc, char* argv[]) {
+                        << " ms (" << 1000.f / wall.count() << " fps)";
+                    cv::putText(curr_frame, out.str(), cv::Point2f(0, 50),
+                                cv::FONT_HERSHEY_TRIPLEX, 0.6, cv::Scalar(0, 0, 255));
+                    if (!isAsyncMode) {  // In the true async mode, there is no way to measure detection time directly
+                        out.str("");
+                        out << "Detection time  : " << std::fixed << std::setprecision(2) << detection.count()
+                        << " ms ("
+                        << 1000.f / detection.count() << " fps)";
+                        cv::putText(curr_frame, out.str(), cv::Point2f(0, 75), cv::FONT_HERSHEY_TRIPLEX, 0.6,
+                            cv::Scalar(255, 0, 0));
+                    }
+                }
+
+                poses = estimator.postprocessCurr();
+
+                if (FLAGS_r) {
+                    sendHumanPose(poses);
+                }
+
+                if (!FLAGS_no_show) {
+                    renderHumanPose(poses, curr_frame);
+                    cv::imshow("ICV Human Pose Estimation on " + FLAGS_d, curr_frame);
+                    t1 = std::chrono::high_resolution_clock::now();
+                    render_time = std::chrono::duration_cast<ms>(t1 - t0).count();
                 }
             }
 
-            if (FLAGS_no_show) {
-                continue;
+            if (isLastFrame) {
+                break;
             }
 
-            renderHumanPose(poses, image);
+            if (isModeChanged) {
+                isModeChanged = false;
+            }
 
-            cv::Mat fpsPane(35, 155, CV_8UC3);
-            fpsPane.setTo(cv::Scalar(153, 119, 76));
-            cv::Mat srcRegion = image(cv::Rect(8, 8, fpsPane.cols, fpsPane.rows));
-            cv::addWeighted(srcRegion, 0.4, fpsPane, 0.6, 0, srcRegion);
-            std::stringstream fpsSs;
-            fpsSs << "FPS: " << int(1000.0f / inferenceTime * 100) / 100.0f;
-            cv::putText(image, fpsSs.str(), cv::Point(16, 32),
-                        cv::FONT_HERSHEY_COMPLEX, 0.8, cv::Scalar(0, 0, 255));
-            cv::imshow("ICV Human Pose Estimation", image);
+            // Final point:
+            // in the truly Async mode we swap the NEXT and CURRENT requests for the next iteration
+            curr_frame = next_frame;
+            next_frame = cv::Mat();
+            if (isAsyncMode) {
+                estimator.swapRequest();
+            }
 
-            int key = cv::waitKey(delay) & 255;
+            const int key = cv::waitKey(delay) & 255;
             if (key == 'p') {
                 delay = (delay == 0) ? 33 : 0;
-            } else if (key == 27) {
+            } else if (27 == key) { // Esc
                 break;
+            } else if (9 == key) { // Tab
+                isAsyncMode ^= true;
+                isModeChanged = true;
+            } else if (32 == key) { // Space
+                FLAGS_black ^= true;
```

Comment:
Please don't modify the `FLAGS_*` variables. It's confusing if `FLAGS_*` doesn't actually contain the value of the corresponding flag.

Reply:
Changed